game/core/src/me/lihq/game/living/NPC.java

```diff
@@ -45,31 +43,70 @@
      *
      * @param tileX       - x coordinate of tile that the NPC will be initially rendered on.
      * @param tileY       - y coordinate of tile that the NPC will be initially rendered on.
-     * @param roomID      - ID of room they are in
+     * @param room        - ID of room they are in
      * @param spriteSheet - Spritesheet for this NPC
      * @param canBeKiller - Boolean whether they can or cannot be the killer
      */
-    public NPC(String name, String spriteSheet, int tileX, int tileY, int roomID, boolean canBeKiller)
+    public NPC(String name, String spriteSheet, int tileX, int tileY, Room room, boolean canBeKiller)
     {
-
-        super(name, spriteSheet, tileX, tileY);
-
-        this.setRoomID(roomID);
-
+        super(name, "npc/" + spriteSheet, tileX, tileY);
+        this.setRoom(room);
+        this.random = new Random();
         this.canBeKiller = canBeKiller;
-
     }
 
 
+    @Override
+    public void update() {
+        super.update();
+        this.randomMove();
+    }
     /**
      * Allow the NPC to move around their room.
-     *
-     * @param dx - how far to move in the x direction
-     * @param dy - how far to move in the y direction
      */
-    public void move(int dx, int dy)
+    public void move(Direction dir)
     {
+        if (this.state != PersonState.STANDING) {
+            return;
+        }
+
+        if (!getRoom().isWalkableTile(this.tileCoordinates.x + dir.getDx(), this.tileCoordinates.y + dir.getDy())) {
+            setDirection(dir);
+            return;
+        }
 
+        if (GameMain.me.player.tileCoordinates.x == this.tileCoordinates.x + dir.getDx() && GameMain.me.player.tileCoordinates.y == this.tileCoordinates.y + dir.getDy())
+        {
+            return;
+        }
+
+        initialiseMove(dir);
+    }
+
+    private void randomMove()
+    {
+        if (getState() == PersonState.WALKING) return;
+
+        if (random.nextDouble() > 0.01) {
+            return;
+        }
+
+        Direction dir;
+
+        Double dirRand = random.nextDouble();
```

Comment:
So why are we having them move in the same direction they last went 50% of the time? Shouldnt we just randomly pick from the 4 directions?